src/app/plugins/ngapimock/ng-apimock-settings.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+
+export interface NgApiMockSettings {
+  integrate: boolean;
+  ngApiMockHost: string;
+  ngApiMockBasePath: string;
+}
+@Injectable({
+  providedIn: 'root',
+})
+export class NgApimockSettingsService {
+  private settings: NgApiMockSettings;
+  private settingKey = 'ngApiMockSettings';
+  constructor() {}
+
+  getSettings(): Promise<NgApiMockSettings> {
+    return new Promise((resolve, reject) => {
+      if (this.settings) {
+        return resolve(this.settings);
+      }
+      chrome.storage.local.get([this.settingKey],
+        (result) => result[this.settingKey] ? resolve(JSON.parse(result[this.settingKey])) : reject('no settings'));
+    });
+  }
+
+  storeSettings(newSettings: NgApiMockSettings): Promise<boolean> {
+    this.settings = newSettings;
+    return new Promise((resolve) => {
+      chrome.storage.local.set({[this.settingKey]:  JSON.stringify(newSettings)}, () => resolve(true));
```

Comment:
No need to stringify/parse. And shouldn't those configs be domain specific (maybe for a later version :)

Reply:
removed stringfy parse. Don't think this need to be specific for a domain. You usually have the same settings for each project. Maybe something for later indeed (overwrite for domain or somehting)